google/cloud/aiplatform/models.py

```diff
@@ -3210,3 +3211,80 @@ def upload_tensorflow_saved_model(
             sync=sync,
             upload_request_timeout=upload_request_timeout,
         )
+
+    def list_model_evaluations(
+        self,
+    ) -> List["model_evaluation.ModelEvaluation"]:
+        """List all Model Evaluation resources associated with this model.
+
+        Example Usage:
+
+        my_model = Model(
+            model_name="projects/123/locations/us-central1/models/456"
+        )
+
+        my_evaluations = my_model.list_model_evaluations()
+
+        Returns:
+            List[model_evaluation.ModelEvaluation]: List of ModelEvaluation resources
+            for the model.
+        """
+        parent = utils.full_resource_name(
+            resource_name=self.resource_name,
+            resource_noun=Model._resource_noun,
+            parse_resource_name_method=Model._parse_resource_name,
+            format_resource_name_method=Model._format_resource_name,
+        )
+
+        self.wait()
+
+        return model_evaluation.ModelEvaluation._list(
+            parent=parent,
+        )
+
+    def get_model_evaluation(
+        self,
+        evaluation_id: Optional[str] = None,
+    ) -> Optional[model_evaluation.ModelEvaluation]:
+        """Returns a ModelEvaluation resource and instantiates its representation.
+        If no evaluation_id is passed, it will return the first evaluation associated
+        with this model.
+
+        Example usage:
+
+            my_model = Model(
+                model_name="projects/123/locations/us-central1/models/456"
+            )
+
+            my_evaluation = my_model.get_model_evaluation(
+                evaluation_id="789"
+            )
+
+            # If no arguments are passed, this returns the first evaluation for the model
+            my_evaluation = my_model.get_model_evaluation()
+
+        Args:
+            evaluation_id (str):
+                Optional. The ID of the model evaluation to retrieve.
+        Returns:
+            model_evaluation.ModelEvaluation: Instantiated representation of the
+            ModelEvaluation resource.
+        """
+        if not evaluation_id:
+            evaluation_resource_name = self.list_model_evaluations()[0].resource_name
```

Comment:
This can throw an exception if there are no model evalutions. Also there is no need to do the additional GET call below:

```Python
evals = self.list_model_evaluations()

if len(evals) > 1:
  _LOGGER.warning(
                f"Your model has more than one model evaluation, this is returning only one evaluation resource: {evals[0].resource_name}"
            )
            
return evals[0] if evals else evals
```

Reply:
Good point re: models w/ no evaluations. I think I've fixed this.